### PR TITLE
[Feature] Add tty option

### PR DESCRIPTION
**Pull request by asaphaaning**
Thanks for branching this wonderful project out into the Rust space!

Some images rely on a tty being available to properly function. This should cover it!

### CODE REVIEW

**Comment by mfelsche** [comment]
Can you provide an example docker image that requires docker to have a pseudo-tty configured? So this can be verified. I am not 100% sure this works with stdout piped https://github.com/testcontainers/testcontainers-rs/blob/dev/testcontainers/src/clients/cli.rs#L188 this is why i ask

**Comment by ASAPHAANING** [comment]
I can't quite reproduce the image (some php + openrc stuff) I was working with but essentially it looked like the snippet below, symlinking from the tty over to the log locations and expecting `/dev/pts/0` to be writable. 

_Example_
```
FROM nginx:stable-alpine

COPY init.sh /

RUN ln -nsf /dev/pts/1 /var/log/nginx/access.log

ENTRYPOINT ["/init.sh"]
```

init.sh
```
chmod o+w /dev/pts/1
```

I tested my changes with the image and confirmed that running with `-it` allows the container to boot.

**Comment by thomaseizinger** [comment]
Thanks! Maintaining a minimal API is something that is really important, hence a couple of questions:

- Would it hurt to always try and allocate a tty and not expose this at all?
- The option seems to be image specific so I think we can remove the override on `RunnableImage`?
- It would be nice to add a test-case with an actual docker image for such purposes. You can add them [here](https://github.com/testcontainers/testcontainers-rs/tree/dev/testimages) so they are available within the test-suite!

**Comment by ASAPHAANING** [comment]
Hi @thomaseizinger!

Thanks for putting this thing together!

I actually do agree that having ptty available is a sensible default. If I flip the bit in the trait default implementation the implementing party will not be required to override it while leaving the the option there if they specifically want to _not_ allocate a ptty. What do you think?

I added a test case too as per request 👍

File: testcontainers/src/core/image.rs [hunk 1]
```diff
@@ -157,6 +162,7 @@ impl ImageArgs for () {
 pub struct RunnableImage<I: Image> {
     image: I,
     image_args: I::Args,
+    tty: bool,
```
Comment (thomaseizinger) [inline testcontainers/src/core/image.rs:165]: Is this necessary? Can't we access it through `image`?

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thank you! I think exposing this makes sense :)

One minor thing.

File: testcontainers/src/core/image.rs (hunk 1 above)
Comment (ASAPHAANING) [reply testcontainers/src/core/image.rs]: Ah, eagle eyes! 🦅  An artifact left from the other review change. It's gone :)

**Comment by ASAPHAANING** [comment]
@thomaseizinger 

Anything else you'd like me to cover? :)

**Review by thomaseizinger — APPROVED** [review]
Yep this looks good. Feel free to `bors r+` at your convenience.

bors d=ASAPHAANING

**Comment by thomaseizinger** [comment]
bors retry

**Comment by ASAPHAANING** [comment]
> bors retry

I think the `test_ipv4_ipv6_host_ports` test is broken on `dev`. I can make the entire suite run if I comment out the last assertion, let me push an example

**Comment by thomaseizinger** [comment]
> > bors retry
> 
> I think the `test_ipv4_ipv6_host_ports` test is broken on `dev`. I can make the entire suite run if I comment out the ipv6 assertions.

Damn. Thanks for investigating.

**Comment by ASAPHAANING** [comment]
bors retry

**Comment by DDtKey** [comment]
We've released a new version of `testcontainers`: [0.16.0](https://github.com/testcontainers/testcontainers-rs/releases/tag/0.16.0).

Could you update and re-open the PR if it's still relevant. 

Closing because of inactivity for now

**Pull request closed without merging**